src/panels/calendar/ha-calendar-app-bar.ts

```diff
@@ -0,0 +1,266 @@
+
+  @property() public params: CalendarAppBarParams = {};
+
+  @property() public navigation?: boolean = true;
+
+  @property() public controls?: boolean = true;
+
+  @property() public refresh?: boolean = true;
+
+  @property({ attribute: false }) public views: FullCalendarView[] = [
+    "dayGridMonth",
+    "dayGridWeek",
+    "dayGridDay",
+    "listWeek",
+  ];
+
+  @property() public initialView: FullCalendarView = "dayGridMonth";
+
+  @state() private _activeView = this.initialView;
+
+  private _viewButtons?: ToggleButton[];
+
+  protected render() {
+    const viewToggleButtons = this._viewToggleButtons(
+      this.views,
+      this.hass.localize
+    );
+    return html`
+      <div class="row">
+        ${this.navigation
+          ? html`
+              <div class="navigation">
+                ${this.label}
+                <ha-icon-button-prev
+                  .label=${this.hass.localize("ui.common.previous")}
+                  class="prev"
+                  @click=${this._handlePrev}
+                >
+                </ha-icon-button-prev>
+                <ha-icon-button-next
+                  .label=${this.hass.localize("ui.common.next")}
+                  class="next"
+                  @click=${this._handleNext}
+                >
+                </ha-icon-button-next>
+                <mwc-button outlined class="today" @click=${this._handleToday}
+                  >${this.hass.localize(
+                    "ui.components.calendar.today"
+                  )}</mwc-button
+                >
+              </div>
+            `
+          : nothing}
+        ${this.controls
+          ? html`
+              <div class="control">
+                <ha-button-toggle-group
+                  .buttons=${viewToggleButtons}
+                  .active=${this._activeView}
+                  @value-changed=${this._handleSelectView}
+                  .dir=${computeRTLDirection(this.hass)}
+                ></ha-button-toggle-group>
+                ${this.refresh
+                  ? html`
+                      <ha-icon-button
+                        .path=${mdiRefresh}
+                        .label=${this.hass.localize("ui.common.refresh")}
+                        @click=${this._handleRefresh}
+                      ></ha-icon-button>
+                    `
+                  : nothing}
+              </div>
+            `
+          : nothing}
+      </div>
+    `;
+  }
+
+  private _handlePrev() {
+    if (this.params.prev) {
+      this.params.prev();
+    }
+  }
+
+  private _handleNext() {
+    if (this.params.next) {
+      this.params.next();
+    }
+  }
+
+  private _handleToday() {
+    if (this.params.today) {
+      this.params.today();
+    }
+  }
+
+  private _handleRefresh() {
+    fireEvent(this, "refresh");
+  }
+
+  private _handleSelectView(ev: CustomEvent): void {
+    ev.stopPropagation();
+    if (this.params.selectView) {
+      this.params.selectView(ev.detail.value);
+    }
+  }
```

Comment:
make these fire events

Reply:
You can create a single helper function that maps the events to the function calls

Reply:
Thanks, fixed. I did have it this way (looks like i left the events already in there!) but then didn't like the manual wiring up of different callbacks. The single helper seems better. I ended up copying between the calendar card and panel which may be ok, but let me know if its worth putting that function somewhere common (i'm not sure where to put it)